The Great Battle Script Cleanup

Pixelstyx opened the pull request:
Every single subscript has had its formatting changed, consts and comments added to better describe what's happening.

Most fixes that were included in this PR were merged elsewhere.

BluRosie committed:
commit work

BluRosie committed:
commit work

Pixelstyx committed:
Merge branch 'move-dev' of https://github.com/BluRosie/hg-engine into move-dev

Pixelstyx committed:
001-022, 362, 364, 455

Pixelstyx committed:
Fixed constant typo

BATTLE_STATUS_SYNCRONIZE -> BATTLE_STATUS_SYNCHRONIZE

Pixelstyx committed:
Up to 026

+tweaks

Pixelstyx committed:
Up to 036

Likely oversight discovered with Dream Eater not having modern interaction with Liquid Ooze.

Pixelstyx committed:
Up to 046, Fix Safeguard preventing status item use, fix Dream Eater ignoring Liquid Ooze

Pixelstyx committed:
0047 + Burn & Poison fixes

Fixed Safeguard still suppressing the effect of held status-inflicting items (changed in Gen V)
Fixed one layer of Toxic Spikes skipping checks for Flower Veil and Misty Terrain
Fixed Flame Orb ignoring Misty Terrain
Fixed Mold Breaker ignoring ally Flower Veil when interacting with Toxic Spikes

BluRosie committed:
make species defines easier for fakemon

BluRosie committed:
celestina work

Pixelstyx committed:
Update subscript_0160_TAILWIND_START.s

Pixelstyx committed:
Clarify subscript 161

Pixelstyx committed:
Clarify subscript 162

Pixelstyx committed:
Clarify subscript 163 and remove vestigial ability checks

Pixelstyx committed:
Clarify subscript 164

Pixelstyx committed:
Clarify subscript 165

Pixelstyx committed:
Clarify subscript 166

Pixelstyx committed:
Clarify subscript 167, remove vestigial ability-based failures

Also modernizes Griseous Orb interaction (as in, it does not).

Pixelstyx committed:
Add Insomnia check to Worry Seed in BeforeMove

Pixelstyx committed:
Clarify subscript 168

Pixelstyx committed:
Add consts for battle_type_11 and the number that kills you instantly

Pixelstyx committed:
Add Role Play to BeforeMove, Fix missing case for Gulp Missile in AbilityFailRolePlay, Fix Belly Drum allowing self-KO in BeforeMove

Pixelstyx committed:
Added missing check for Arceus plates to CanItemBeRemovedFromSpecies

Pixelstyx committed:
Up to 158, except for 88 & 91

Pixelstyx committed:
Merge branch 'move-dev-cleanup' of https://github.com/Pixelstyx/hg-engine into move-dev-cleanup

Pixelstyx committed:
Clarified subscript 159

Pixelstyx committed:
Up to end of vanilla scripts

Fixed Griseous Orb and Pokemon with Multitype being impervious to Embargo (changed in Gen V).

Pixelstyx committed:
Clarified MOVE_470 as MOVE_MEGA_EVOLVE

Pixelstyx committed:
Clarify all remaining subscripts

Pixelstyx committed:
Clarify 88 and 91

Pixelstyx committed:
Fix Roseli Berry message checking Chill Drive

Pixelstyx committed:
Fix Moody using ability const for Sand Veil

Pixelstyx committed:
Game stat consts

Pixelstyx committed:
Clarify TryPartyStatusRefresh in 088

Pixelstyx committed:
Clarify where some scripts are called

Pixelstyx committed:
Add Stat consts

Pixelstyx committed:
Formatting fixes

Pixelstyx committed:
Fix Color Change, Soak, Magic Powder not removing type3 and allow Soak animation to play even when types cause failure

Also sets up Magic Powder to allow the same animation conditions, but does not enable it as more testing is needed.

Pixelstyx committed:
Fixed Weakness Policy calling the message for ally Sweet Veil and solely raising Attack by 1 stage.

Pixelstyx committed:
Fixed Shell Smash aggregating maxed stats into a single message rather than listing them individually.

Pixelstyx committed:
Fixed Spicy Extract lacking individualized messages for Attack and Defense when one or both cannot be changed further.

Pixelstyx committed:
Removed Shadow Force check from Shed Tail.

Pixelstyx committed:
Merge branch 'move-dev' into move-dev-cleanup

Pixelstyx committed:
Hook CheckToxicSpikes to allow type3 and tera Poison to remove Toxic Spikes

Pixelstyx committed:
Merge branch 'move-dev-cleanup' of https://github.com/Pixelstyx/hg-engine into move-dev-cleanup

Pixelstyx committed:
Fixed form changes not reverting type3.

Pixelstyx committed:
Replace _000 with _Start in every move script

Pixelstyx committed:
Modernize every effect script

Modernized Fling behaviour to allow the Griseous Orb to be flung and for the move to work when used by Pokemon with Multitype (Arceus).

Fixed Judgment and Multi-Attack calling TYPE_MYSTERY for Pixie Plate instead of TYPE_FAIRY (both have an ID of 9, though in some cases the latter is different).

Fixed Acrobatics calling BSCRIPT_VAR_BATTLE_TYPE instead of ITEM_NONE (both have an ID of 0)
Fixed typo in the name of effect script 318 (Geomancy) that indicated it raised Atk instead of SpDef.

Fixed oversight in effect script 321 (Infernal Parade) where Comatose was not being checked.

Pixelstyx committed:
Every missing contest type and move description

Pixelstyx committed:
Appeal effect consts

Pixelstyx committed:
Missed a spot

Pixelstyx committed:
Fixed comma on SELF_KO_DAMAGE const

Pixelstyx committed:
Removed unnecessary part of game stats header file

Pixelstyx committed:
Revert change to name of Max Hailstorm

Pixelstyx committed:
Fixed Darkest Lariat not ignoring Defense

Pixelstyx committed:
Mark all Max Moves as unimplemented.

Pixelstyx committed:
More comments

Pixelstyx committed:
seriously what even are these

Pixelstyx committed:
lil fixes

mozzydippers commented on data/battle_scripts/effects/effect_script_0010_ATK_UP.s:
now only called by sharpen since howl got moved into its own subscript bc of modernization & such

Pixelstyx replied:
Fixed, forgot to go back for that one.

mozzydippers commented on data/battle_scripts/effects/effect_script_0178_COPY_ABILITY.s:
most of the checks in role play skill swap etc can be removed in cleanup given theyre present in beforemove

Pixelstyx replied:
I was advised by Blu not to touch 99% of fail checks to not even further complicate this pass. We'll handle them after for certain.

Pixelstyx committed:
Fixed calling comment for atk up subscript

Howl is now handled elsewhere.

Pixelstyx commented on data/battle_scripts/effects/effect_script_0103_PRIORITY_1.s:
```diff
@@ -2,7 +2,9 @@
 
-_000:
+// Called by Accelerock, Aqua Jet, Bullet Punch, Grassy Glide, Ice Shard, Jet Punch, Mach Punch, Quick Attack, Shadow Sneak and Vacuum Wave.
+// ...for some reason.
```
This is what I am gathering from reading some other spots in the code-- it seems like eff IDs are used elsewhere in the code to check whether things like Substitute should work. Until we know where, though, I don't want to pretend we do.

Pixelstyx commented on data/battle_scripts/effects/effect_script_0111_PROTECT.s:
```diff
@@ -2,6 +2,7 @@
 .data
 
-_000:
+// Called by Protect and Detect.
```
For indexing purposes I try not to namedrop moves not called by an effect script. If we're being thorough and the others aren't handled here, it should be implicit. Also, the user should be able to look up King's Shield or Baneful Bunker and figure out where it is (or isn't)

Pixelstyx commented on data/battle_scripts/effects/effect_script_0140_RAISE_ALL_STATS_HIT.s:
```diff
@@ -2,7 +2,8 @@
 .data
 
-_000:
+// Called by Ancient Power, Ominous Wind and Silver Wind.
```
It might end up being its own effscript since it needs to do other things to 'use up' the z-crystal, but we could comment it the way I did with a few G-Max Moves to say we're not sure yet.

Pixelstyx commented on data/battle_scripts/effects/effect_script_0198_RECOIL_THIRD.s:
```diff
@@ -2,11 +2,12 @@
 
-_000:
-    CheckAbility CHECK_OPCODE_NOT_HAVE, BATTLER_CATEGORY_ATTACKER, ABILITY_RECKLESS, _008
+// Called by Brave Bird, Double-Edge, Flare Blitz, Volt Tackle, Wave Crash and Wood Hammer.
```
Oops. Those will be fixed in the upcoming commit. When I first got here it hadn't quite sunk in how eff scripts work and I hadn't returned to it.

Pixelstyx committed:
Fixed a few calling comments

Added Hold Back & Clangorous Soulblaze and removed incorrect use of Flare Blitz & Volt Tackle.

Pixelstyx committed:
Fixed missing protect variants in calling comments.

Pixelstyx committed:
Added battle type defines for parity with asm

Pixelstyx committed:
Fixed missing changes to subscript 5

They were in the wrong PR, oops!

Pixelstyx committed:
Formatting & Typo fixes, demystify BATTLE_TYPE_13

Pixelstyx committed:
Merge remote-tracking branch 'upstream/main' into move-dev-cleanup

Pixelstyx committed:
Revert "celestina work"

This reverts commit 91c621656d4d8af087c57538b03c3a8d74b8001b.

Pixelstyx committed:
Assorted cleanup

Pixelstyx committed:
Clean up new scripts

Pixelstyx committed:
Remove wondering

Pixelstyx committed:
Add Tera todos to effect scripts
